### PR TITLE
Run Odo registry command test with OCI-based staging repository

### PR DESCRIPTION
**Pull request by josephca**
### What does this PR do?
Run Odo integration tests with OCI-based staging repository

### CODE REVIEW

**Review by johnmcollier — COMMENTED** [review]
Before the test suites you need to explicitly make sure that you remove the `DefaultDevfileRegistry` entry that points to https://registry.devfile.io. Basically something like

```
odo registry delete DefaultDevfileRegistry -f
odo registry add TestDevfileRegistry https://registry.stage.devfile.io
```

**Comment by josephca** [comment]
> Before the test suites you need to explicitly make sure that you remove the `DefaultDevfileRegistry` entry that points to https://registry.devfile.io. Basically something like
> 
> ```
> odo registry delete DefaultDevfileRegistry -f
> odo registry add TestDevfileRegistry https://registry.stage.devfile.io
> ```

@johnmcollier Good point! I'll add the steps. Thanks,

File: tests/integration/devfile/devfile_suite_test.go [hunk 1]
```diff
@@ -13,9 +13,8 @@ func TestDevfiles(t *testing.T) {
 
 var _ = BeforeSuite(func() {
 	const registryName string = "DefaultDevfileRegistry"
-	// Use staging OCI-based registry for tests to avoid a potential overload
 	const addRegistryURL string = "https://registry.stage.devfile.io"
 
-	helper.CmdShouldPass("odo", "registry", "delete", "DefaultDevfileRegistry", "-f")
-	helper.CmdShouldPass("odo", "registry", "add", registryName, addRegistryURL)
+	// Use staging OCI-based registry for tests to avoid a potential overload
+	helper.CmdShouldPass("odo", "registry", "update", registryName, addRegistryURL)
```
Comment (johnmcollier) [inline tests/integration/devfile/devfile_suite_test.go:19]: FYI, this approach won't work I don't think:

As part of the migration from Github-based registries to registry.devfile.io, Odo has logic in it to detect if `DefaultDevfileRegistry` points to a URL other than `https://registry.devfile.io` and reverts it back accordingly if it differs

```
johncollier@jcollier-mac ~ % odo registry list
NAME                       URL                             SECURE
DefaultDevfileRegistry     https://registry.devfile.io     No
johncollier@jcollier-mac ~ % odo registry update DefaultDevfileRegistry https://registry.stage.devfile.io
? Are you sure you want to update registry "DefaultDevfileRegistry" Yes
Successfully updated registry
johncollier@jcollier-mac ~ % odo registry list
NAME                       URL                             SECURE
DefaultDevfileRegistry     https://registry.devfile.io     No
```

**Comment by josephca** [comment]
Discovered an issue and this PR depends on fix - https://github.com/openshift/odo/pull/4702

File: tests/integration/devfile/devfile_suite_test.go (hunk 1 above)
Comment (johnmcollier) [reply tests/integration/devfile/devfile_suite_test.go:19]: @josephca Ahh, just saw the issue Jingfu opened. Nevermind my comment then, once that is fixed in Odo, this should work.

Comment (josephca) [reply tests/integration/devfile/devfile_suite_test.go:19]: @johnmcollier  Yup. That's why I added a comment below. :)

**Comment by josephca** [comment]
This PR is replaced by https://github.com/devfile/integration-tests/pull/21 which targets the new 'main' branch.

**Pull request closed without merging**